pkg/etcd/etcd.go

```diff
@@ -414,11 +423,7 @@ func (e *ETCD) setName(force bool) error {
 	fileName := nameFile(e.config)
 	data, err := ioutil.ReadFile(fileName)
 	if os.IsNotExist(err) || force {
-		h, err := os.Hostname()
-		if err != nil {
-			return err
-		}
-		e.name = strings.SplitN(h, ".", 2)[0] + "-" + uuid.New().String()[:8]
+		e.name = strings.SplitN(e.config.ServerNodeName, ".", 2)[0] + "-" + uuid.New().String()[:8]
```

Comment:
I know it's appending a uuid, but does this assume short hostnames are globally unique? What about:
```
server-1.zone-a.cloud
server-2.zone-a.cloud
server-1.zone-b.cloud
server-2.zone-b.cloud
```

Is there any reason the member name can't just be the full node name or the Node object uid or something like that?

Reply:
I didnt want to change the old behavior but you are 100% right, we shouldnt just shorten the name

Reply:
I will fix

Reply:
The etcd member management controller/annotation handling is built to expect unique etcd node names. If they are exactly hostname, this will break my controller and other assumptions down the line, so I would prefer we keep the existing behavior of appending part of a uuid to the end of the etcd member name.

Reply:
Also, this seems to change behavior. Before, an etcd member name was always based on the `os.Hostname()`, now, it defaults to `os.Hostname()` but can be explicitly overwritten by setting the node name via CLI flag.

Not sure if this should affect anything, I don't think it does, on existing nodes that specify a nodename via config, they will continue to use the old etcd member name as defined in the file until the next time they tombstone, at which point their etcd member name will reflect their chosen node name

Reply:
I wasnt planning to remove the uuid part, but we do have a larger problem @brandond @Oats87 :

If the user didnt specify the node-name, it will default to the os.Hostname, which doesnt get the fqdn anyway, so we will end up with a problem of duplicate nodes

So we can either move to the use of FQDN over the Hostname, but this will break the upgrades of k3s/rke2, or we can just document this part? any thoughts

Reply:
> If the user didnt specify the node-name, it will default to the os.Hostname, which doesnt get the fqdn anyway, so we will end up with a problem of duplicate nodes

As long as we have the same issue with duplicate names for the kubernetes Node object we should be fine. The user will figure it out pretty quick if their nodes fail to register with the cluster due to conflicting node names.